draft-ietf-deleg.md

```diff
@@ -264,61 +264,55 @@ Rest of the Step 2's description is not affected by this document.
 Please note the instructions to "Bound the amount of work" further down in the original text to apply. Suitable limits MUST be enforced to limit damage EVEN IF SOMEONE HAS INCORRECTLY CONFIGURED SOME DATA.
 
 ## Authoritative Servers
-DELEG aware authoritative servers have to act different when handling DE=0 queries from DELEG unaware clients, and DE=1 queries from DELEG aware clients.
+DELEG-aware authoritative servers act differently when handling queries from DELEG-unaware clients (those with DE=0) and queries from DELEG-aware clients (those with DE=1).
 
-### Unaware clients, DE=0
-A query with DE EDNS bit cleared indicates a DELEG unaware client. Such client has no use for DELEG information. In this case DELEG aware authoritative server MUST answer as if it is DELEG unaware.
+DNSSEC signing in the presence of DELEG records is unaffected.
+That is, the presence or abscence of DELEG records is always reflected in the NSEC type bitmap, regardless of the type of query.
 
-In summary, in this specific case, DELEG RR does not create zone cut and is not returned in referral responses, and is not considered authoritative on the parent side of a zone cut. DNSSEC signing is unaffected by this instruction, i.e. no special handling for NSEC type bitmap is necessary.
-
-Two corner cases arise if a zone cut contains DELEG RR and server is answering a DE=0 query:
-
-#### QTYPE=DELEG
-
-DE=0 query indicates that for this particular client DELEG RR type is 'not authoritative' on the parent side.
-
-Let zone file for root zone contain following delegation with NS and DELEG records:
-
-    example. NS    ...
-    example. DELEG ...
+The server MUST copy the value of the DE bit from the query into the response.
+(TODO: not really necessary protocol-wise, but might be nice for monitoring the deployment?)
 
-In this specific case, explicit QTYPE=DELEG query will result in referral response generated by following DELEG unaware rules:
+### DELEG-unaware Clients
 
-    ;; Authority
-    example. NS    ...
+DELEG-unaware clients do not use DELEG records for delegation.
+When a DELEG-aware authoritative server responds to a DELEG-unaware client, any DELEG RR in the response does not create zone cut, is not returned in referral responses, and is not considered authoritative on the parent side of a zone cut.
+Because of this, DELEG-aware authoritative servers MUST answer as if they are DELEG-unaware except for two narrow cases described here.
 
-Other RRs might be included as specified by pre-DELEG protocols.
+#### DELEG-unaware Clients Requesting QTYPE=DELEG
 
-#### DELEG without NS
+In DELEG-unaware clients, records with the DELEG RRtype are not authoritative on the parent side.
+Thus, queries with DE=0 and QTYPE=DELEG MUST result in a legacy referral response.
```

Comment:
Does 'legacy referral response' need to be defined?

Reply:
Well, "legacy" is defined earlier in the document, so I don't think so.